PSEUDO API 1.md

```diff
@@ -0,0 +1,166 @@
+
+## XR
+
+	interface XR {
+		Promise<sequence<XRDisplay>> getDisplays();
+	};
+
+
+## XRDisplay
+
+	interface XRDisplay {
+		readonly attribute DOMString displayName;
+		readonly attribute boolean isExternal;
+
+		Promise<boolean> supportsSession(XRSessionCreateParametersInit parameters);
+		Promise<XRSession> requestSession(XRSessionCreateParametersInit parameters);
+
+		attribute EventHandler ondeactivate;
+	};
+
+### Todo
+
+- calibration
+
+## Reality
+
+	interface Reality {
+		readonly atttribute DOMString realityName;
+		readonly attribute boolean hasPointCloud;
+		readonly attribute XRCoordinates stageLocation;
+
+		Promise<boolean> requestStageLocation(float x, float y float z, XRCoordinates)
+		Promise<boolean> requestResetStageLocation()
+
+		Promise<XRPointCloud> getPointCloud()
+
+		Promise<XRLightEstimate> getLightEstimate()
+
+		attribute EventHandler onchange;
+	};
+
+### Todo
+
+- configuration
+- anchors
+- manifold instead of point cloud
+
+## XRPointCloud
+
+	interface XRPointCloud {
+		readonly attribute Float32Array points;
+	}
+
+## XRLightEstimate
+
+	interface XRLightEstimate {
+		readonly attribute float ambientIntensity;
+		readonly attribute float ambientColorTemperature;
+	}
+
+## XRSession
+
+	interface XRSession : EventTarget {
+		readonly attribute XRDisplay display;
+		readonly attribute XRSessionCreateParameters createParameters;
+
+		attribute double depthNear;
+		attribute double depthFar;
+
+		attribute XRLayer layer;
+
+		Promise<sequence <Reality>> getRealities();
+
+		Promise<XRFrameOfReference> requestFrameOfReference(XRFrameOfReferenceType type);
+
+		long requestFrame(XRFrameRequestCallback callback);
+		void cancelFrame(long handle);
+
+		Promise<void> endSession();
+
+		attribute EventHandler onblur;
```

Comment:
In WebVR, these blur and focus events are for when someone switches away from the tab or window, but the layer could also have similar events as the user focuses individual layers.